stripped/colabs/Zbiory_treningowe_i_testowe_stripped.py

```diff
@@ -0,0 +1,384 @@
+"""
+
+"""
+Poglądowa grafika przedstawiająca obszar zbioru testowego:
+![testset.png](images/testset.png)
+"""
+
+rectangle1 = get_rectangle_coordinates([-122,25.5], [-115, 37], orientation=True)
+rectangle2 = get_rectangle_coordinates([-124,37], [-122, 52], orientation=True)
+rectangle3 = get_rectangle_coordinates([-93, 24.5], [-79.5, 34], orientation=True)
+rectangle4 = get_rectangle_coordinates([-79.5, 34], [-75, 37], orientation=True)
+rectangle5 = get_rectangle_coordinates([-115, 25.5], [-111, 30], orientation=True)
+
+test_set_coordinates = rectangle1 + rectangle2 + rectangle3 + rectangle4 + rectangle5
+len(test_set_coordinates)
+
+
+train_set_coordinates = coordinates.values.tolist()
+for t in  test_set_coordinates:
+   train_set_coordinates.remove(t)
+
+
+test_set_coords = pd.DataFrame(test_set_coordinates, columns=['lon', 'lat'])
+train_set_coords = pd.DataFrame(train_set_coordinates, columns=['lon', 'lat'])
+
+
+spark.conf.set("spark.sql.execution.arrow.pyspark.enabled", "true")
+test_df = spark.createDataFrame(test_set_coords)
+train_df = spark.createDataFrame(train_set_coords)
+
+
+cond = [train_df['lon'] == nasa['lon'], train_df['lat'] == nasa['lat']]
+train_set = nasa.join(train_df, cond, 'inner').select(nasa['*']).show()
+
+
+cond = [test_df['lon'] == nasa['lon'], test_df['lat'] == nasa['lat']]
+test_set = nasa.join(test_df, cond, 'inner').select(nasa['*']).show()
+
+
+"""
+**Drugi wariant:**  spatial block cross-validation, którą możemy spróbować delikatnie zmodyfikować dodając za każdym razem do *foldu* testowego uprzednio "wycięty" fragment pustyni (być może *Baja California*?)
+"""
+
+"""
+Poniższy kod dzieli współrzędne na kwadratowe bloki (generuje grid) o zadanej wielkości:
+"""
+
+import math
+
+#grid_cell_size - the grid cell size (the length of the side of a square) in degrees
+def get_grid(grid_cell_size):
+
+ min_lat = 25.0625
+ max_lat = 52.9375
+ min_lon = -124.9375
+ max_lon = -67.0625
+
+ area = (max_lat-min_lat)*(max_lon-min_lon)
+ cells_num = area//(grid_cell_size*grid_cell_size)
+ actual_grid_size = math.sqrt(area/cells_num)
+
+ xx=np.arange(min_lon, max_lon, step = actual_grid_size)
+ yy=np.arange(min_lat, max_lat, step = actual_grid_size)
+ #xx, yy = np.meshgrid(x,y, sparse=True)
+
+ return(xx, yy)
+
+
+#coordinates: pandas DataFrame with "lon" and "lat" columns containing coordinates
+#block_size: approximate size of a block
+def block_partition(coordinates, block_size):
+
+  coords = coordinates
+  blocks = []
+  xx, yy = get_grid(block_size)
+
+  for y in range(len(yy)-1):
+      for x in range(len(xx)-1):
+          block = []
+          for c in coords.index:
+               lon = coords.at[c,'lon']
+               lat = coords.at[c, 'lat']
+               if (lon >= xx[x])&(lon <= xx[x+1]):
+                   if (lat >= yy[y])&(lat < yy[y+1]):
```

Comment:
`if (xx[x] <= lon <= xx[x+1]) and (yy[y] <= lat < yy[y+1]):`
czy tam miało być `<`

Reply:
not relevant anymore